internal/daemon/controller/handlers/credentiallibraries/credentiallibrary_service.go

```diff
@@ -138,20 +138,33 @@ func NewService(
 	if repoFn == nil {
 		return Service{}, errors.New(ctx, errors.InvalidParameter, op, "missing vault credential repository")
 	}
-	if baseCredentialLibraryRepoFn == nil {
-		return Service{}, errors.New(ctx, errors.InvalidParameter, op, "missing generic credential library repository")
-	}
 	if maxPageSize == 0 {
 		maxPageSize = uint(defaultMaxPageSize)
 	}
 	return Service{
-		iamRepoFn:                   iamRepoFn,
-		repoFn:                      repoFn,
-		baseCredentialLibraryRepoFn: baseCredentialLibraryRepoFn,
-		maxPageSize:                 maxPageSize,
+		iamRepoFn:   iamRepoFn,
+		repoFn:      repoFn,
+		maxPageSize: maxPageSize,
 	}, nil
 }
 
+type credentialLibraryRepo struct {
+	repo *vault.Repository
+}
+
+func (clr *credentialLibraryRepo) GetTotalItems(ctx context.Context) (int, error) {
+	return credential.GetTotalCredentialLibraries(ctx, clr.repo)
+}
+
+func (clr *credentialLibraryRepo) ListDeletedIds(ctx context.Context, since time.Time) ([]string, error) {
+	return credential.ListDeletedCredentialLibraryIds(ctx, nil /* What do here? */, clr.repo, since)
```

Comment:
This function takes a db.Writer, which isn't something the handler generally has access to. How should we pass it in here?